subsys/fs/nvs/nvs.c

```diff
@@ -1346,3 +1346,40 @@ ssize_t nvs_calc_free_space(struct nvs_fs *fs)
 	}
 	return free_space;
 }
+
+size_t nvs_sector_max_data_size(struct nvs_fs *fs)
+{
+	size_t ate_size;
+
+	if (!fs->ready) {
+		LOG_ERR("NVS not initialized");
+		return -EACCES;
+	}
+
+	ate_size = nvs_al_size(fs, sizeof(struct nvs_ate));
+
+	return fs->ate_wra - fs->data_wra - ate_size;
```

Comment:
we should take into account as well the NVS_DATA_CRC_SIZE, right ?

Reply:
Indeed, this is a nice spot !
Fixed.